cgi-bin/LJ/Web.pm

```diff
@@ -2738,19 +2738,47 @@ sub control_strip
                     'trusted_by'        => BML::ml('web.controlstrip.status.trustedby', {'user' => $journal_display}),
                     'watched_by'        => BML::ml('web.controlstrip.status.watchedby', {'user' => $journal_display}),
                     );
-    # Style the status text
-    foreach my $key (keys %statustext) {
-        $statustext{$key} = "<span id='lj_controlstrip_statustext'>" . $statustext{$key} . "</span>";
```

Comment:
It looks like this `foreach` loop is intended to allow multiple possible statustext lines which are appended to the output in sequence, but in your rewrite, you only assign one line value to `$template_args->{'statustext'}`. While a single line is definitely the most common case, I believe it is possible to for multiple lines to be returned here, and we should continue to support that.

Reply:
I analyzed the logic pretty carefully, and I'm like 98% sure it's intended as one-and-only-one. 

Because of how tangled it is, you pretty much have to look for every instance of `$ret .= $statustext{'<SOMETHING>'};` and keep track of the nested if/elsif blocks until your eyes cross, but I did that several times and I _think_ there are no gaps. Also, the _extremely_ granular statustext options are another argument for that interpretation — if we had multiple chances to add to the status text, we probably wouldn't be making a distinction between `mutualtrust_watch` and `mutualtrust_mutualwatch`. 

I think the foreach loop is just laziness — they might only use one statustext per pageload, but within the logic they end up calling `$ret .= $statustext{'<ETC>'};` like 35 times or something, and wrapping the span around it each time would have been demented. So I just moved it to the template.

Anyway, if y'all decide it's necessary we can keep an array of statustexts (like with actionlinks etc.) and just do a foreach in the template instead. But if my reading of the conditionals is right, this is a chance to clearly express an intent that was impossible to express in the old format.

Reply:
Sure, that sounds fair. We can always adjust it later if needed.